app/services/typeahead_service.rb

```diff
@@ -6,98 +6,90 @@
 # TODO: Refactor TypeaheadService and SearchService so that common functions come from a higher level class?
 class TypeaheadService
   def initialize; end
+  include SolrToggle
 
-  def users(params, limit)
-    @users ||= find_users(params, limit)
-  end
-
-  def tags(params, limit)
-    @tags ||= find_tags(params, limit)
-  end
-
-  def notes(params, _limit)
-    @notes ||= find_notes(params)
-  end
-
-  def wikis(params, _limit)
-    @wikis ||= find_wikis(params)
-  end
+  # search_users() returns a standard TagResult; 
+  # users() returns an array of User records
+  # It's unclear if TagResult was supposed to be broken into other types like DocResult?
+  # but perhaps could simply be renamed Result.
 
-  def maps(params, limit)
-    @maps ||= find_maps(params, limit)
+  def users(input, limit = 5)
+    User.limit(limit)
+        .order('id DESC')
+        .where('username LIKE ? AND status = 1', '%' + input + '%')
   end
 
-  def comments(params, _limit)
-    @comments ||= find_comments(params)
-  end
-
-  def find_users(input, limit = 5)
-    DrupalUsers.limit(limit)
-               .order('uid DESC')
-               .where('name LIKE ? AND access != 0', '%' + input + '%')
-  end
-
-  def find_tags(input, limit = 5)
+  def tags(input, limit = 5)
     Tag.includes(:node)
        .where('node.status = 1')
        .limit(limit)
        .where('name LIKE ?', '%' + input + '%')
   end
 
-  def find_comments(input, limit = 5)
+  def comments(input, limit = 5)
     Comment.limit(limit)
            .order('nid DESC')
            .where('status = 1 AND comment LIKE ?', '%' + input + '%')
   end
 
-  ## search for node title only
-  ## FIXme with solr
-  def find_notes(input, limit = 5)
-    Node.limit(limit)
-        .order('nid DESC')
-        .where('type = "note" AND node.status = 1 AND title LIKE ?', '%' + input + '%')
+  def notes(input, limit = 5)
+    if solrAvailable
```

Comment:
Can we access the `params` object in a service, here? If so, we could check for something like `?solr=true` here.

Reply:
I would guess it can't access `params`. But we can test this out in unstable so i guess that'd be OK.